libethereum/EthereumCapability.cpp

```diff
@@ -492,93 +507,65 @@ vector<NodeID> EthereumCapability::selectPeers(
     return allowed;
 }
 
-std::pair<std::vector<NodeID>, std::vector<NodeID>> EthereumCapability::randomPartitionPeers(
-    std::vector<NodeID> const& _peers, std::size_t _number) const
+std::vector<NodeID> EthereumCapability::randomPeers(
+    std::vector<NodeID> const& _peers, size_t _count) const
 {
-    vector<NodeID> part1(_peers);
-    vector<NodeID> part2;
-
-    if (_number >= _peers.size())
-        return std::make_pair(part1, part2);
-
-    std::shuffle(part1.begin(), part1.end(), m_urng);
-
-    // Remove elements from the end of the shuffled part1 vector and move them to part2.
-    std::move(part1.begin() + _number, part1.end(), std::back_inserter(part2));
-    part1.erase(part1.begin() + _number, part1.end());
-    return std::make_pair(move(part1), move(part2));
+    std::vector<NodeID> peers{_peers};
+    if (peers.empty())
+        return peers;
+
+    if (_count > peers.size())
```

Comment:
I was thinking that there could be some benefit to this, but after thinking about it some more there isn't any - the order in which we send blocks to peers doesn't matter since they get all of the new blocks anyway.